app/models/user.rb

```diff
@@ -26,4 +26,9 @@ class User < ApplicationRecord
 
   has_many :reports, dependent: :destroy
   has_many :periods, through: :reports
+
+  def current_report
+    current_report = Report.joins(:period).where(periods: {current: true})
+    return current_report
+  end
```

Comment:
The way you have it now, `User#current_report` doesn't actually limit the reports that are returned to user that is calling this method. It will return a collection of **all** the reports that are current, which I don't think is what you want this method to do.

If you want to do something like `current_user.current_report` and get the current report for `current_user`, you can leverage the nice scope you made in the Report model.

```suggestion
  def current_report
    return reports.current_report.first
  end
```

How does that sound?